src/BusinessChat/BCChatAction.cs

```diff
@@ -0,0 +1,15 @@
+#if XAMCORE_2_0
+using System;
+using System.Collections.Generic;
+using XamCore.Foundation;
+
+namespace XamCore.BusinessChat {
+	public partial class BCChatAction {
+		public static void OpenTranscript (string businessIdentifier, Dictionary<BCParameterName, string> intentParameters) {
+			var keys = Array.ConvertAll<string, NSString> (new List<string>(intentParameters.Values).ToArray (), v => new NSString (v));
+			var values = Array.ConvertAll<BCParameterName, NSString> (new List<BCParameterName> (intentParameters.Keys).ToArray (), k => k.GetConstant ());
+			OpenTranscript (businessIdentifier, NSDictionary<NSString, NSString>.FromObjectsAndKeys (values, keys));
```

Comment:
I'm not sure how many items are expected in the dictionary...
Maybe create a `MutableDictionary` and iterating/adding the items is better
otherwise having two array is an option (single call to native `Dictionary`) but let's get away from the temporary lists

Reply:
I would go one step further, there is no need for the managed representation of a `NSString` so I would create an array of IntPtrs with`NSString.CreateNative` and create the NSMutableArray from those instead just don't forget to release them with `NSString.ReleaseNative`